assets/javascripts/modules/autoComplete.js

```diff
@@ -314,28 +323,99 @@ var suggestionsEvent = function () {
   });
 };
 
+
+/**
+  * Generate random variable name to ensure no conflicts
+*/
+var randomVariableName = function () {
```

Comment:
IMO guarding against a conflict with creating a random global variable is hiding the issue away from the developer. If the global is being overridden due to a variable of the same name already existing the code shouldn't abstract that away and guard for it, the developer should change the name of the global to fix the problem at the source.